### ISSUE TITLE
False positive with NP_NONNULL_PARAM_VIOLATION and Spring RestTemplate HttpEntity

### ISSUE THREAD

**Issue by a-stewart** [spotbugs/spotbugs#1087]
Using Spotbugs 3.1.12.

It is giving a NP_NONNULL_PARAM_VIOLATION for this code: `new HttpEntity<>(null);`

However, it is happy with: `new HttpEntity<>(null, null)`.

This is using Spring Web 5.2.3.
Here is the code in the Spring library (decompiled by IntelliJ):

```java
	/**
	 * Create a new {@code HttpEntity} with the given headers and no body.
	 * @param headers the entity headers
	 */
	public HttpEntity(MultiValueMap<String, String> headers) {
		this(null, headers);
	}

	/**
	 * Create a new {@code HttpEntity} with the given body and headers.
	 * @param body the entity body
	 * @param headers the entity headers
	 */
	public HttpEntity(@Nullable T body, @Nullable MultiValueMap<String, String> headers) {
		this.body = body;
		HttpHeaders tempHeaders = new HttpHeaders();
		if (headers != null) {
			tempHeaders.putAll(headers);
		}
		this.headers = HttpHeaders.readOnlyHttpHeaders(tempHeaders);
	}
```

The bug description is:
This method passes a null value as the parameter of a method which must be non-null. Either this parameter has been explicitly marked as @Nonnull, or analysis has determined that this parameter is always dereferenced.

The method is not marked as @Nonnull and there is a nullcheck on the headers.

This isn't an issue for us as we can use HttpEntity.EMPTY, but thought I'd raise it in case the issue is present somewhere else.

MCVE:

```java
import org.springframework.http.HttpEntity;

class SpotbugsIssue {
  void spotbugsIssue() {
    var x = new HttpEntity<>(null);
  }
}
```

**Comment by ThrawnCA** [comment]
If you check the `package-info.java` file for that package, you'll find that it has annotations indicating that all parameters should be non-nullable by default.

**Comment by a-stewart** [comment]
Thanks, I guess this is a bug that needs raising with the Spring devs then
